src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let numbers = [];
+
+  const adder = (number) => {
+    if (number !== undefined) {
+      numbers.push(number);
+
+      return adder;
+    } else {
+      const result = numbers.reduce((sum, num) => sum + num, 0);
```

Comment:
Nice variant. But there you make unnecessary iterations. You can create number variable instead numbers and update a value